eng/pipelines/templates/steps/vmr-prepare.yml

```diff
@@ -28,11 +33,12 @@ steps:
     path: vmr
     clean: true
 
-- script: |
-    git checkout --track origin/${{ parameters.vmrBranch }}
-    echo "##vso[task.setvariable variable=vmrBranch]${{ parameters.vmrBranch }}"
-  displayName: Check out ${{ parameters.vmrBranch }}
-  workingDirectory: $(Agent.BuildDirectory)/vmr
+- ${{ if parameters.checkoutToBranch }}:
```

Comment:
Please double check that we really only need to checkout for the vmr-synchronization job. @premun PTAL

Reply:
So, we definitely need to for PRs. PRs are run from a branch that does not exist in the VMR. This means that when we do `checkout` of the VMR, we cannot use that branch because it wouldn't exist. That means, we need to check out `main`, because we need something that exists, and then we need to switch to the target PR branch.

The worst part is that we could just check out `$(System.PullRequest.targetBranchName)` but that variable is not available soon enough, only on runtime :-/

Reply:
But only for PRs from installer, right? Inside dotnet/dotnet we don't need to checkout anything?

Reply:
Inside `dotnet/dotnet` we don't run the synchronization

Reply:
But inside `dotnet/dotnet` we still need to give it some branch to check out the VMR at so the `checkout` must be different then. Not sure if it is.

Reply:
Yes, that's the point. Do we need any special logic inside dotnet/dotnet or can we just rely on the default AzDO behavior that clones and checks-out the corresponding SHA?

Reply:
Yeah, we should possibly just do `if not(parameters.isBuiltFromVmr)` around the checkout step

Reply:
But only if the `checkout` task really clones at the right SHA

Reply:
> But only if the checkout task really clones at the right SHA

How could that not be the case inside dotnet/dotnet?

Reply:
https://github.com/dotnet/installer/blob/156efab2e4f52d0586d16dad8127af3f71d6fe4b/eng/pipelines/templates/jobs/vmr-build.yml#L131-L137 only runs for installer.

https://github.com/dotnet/installer/blob/156efab2e4f52d0586d16dad8127af3f71d6fe4b/eng/pipelines/vmr-build.yml#L42-L49 only runs for installer as well.

My assumption is that inside dotnet/dotnet the YMLs don't need vmr-prepare.yml (nor the vmr-pull-update.yml but that's already the case): https://github.com/dotnet/installer/blob/156efab2e4f52d0586d16dad8127af3f71d6fe4b/eng/pipelines/templates/jobs/vmr-build.yml#L122-L137

But that would mean that we wouldn't clone into the "vmr" subfolder anymore and I'm not sure if that's fine? Also, we would need to move the CG task out: https://github.com/dotnet/installer/blob/156efab2e4f52d0586d16dad8127af3f71d6fe4b/eng/pipelines/templates/steps/vmr-prepare.yml#L37-L43

So maybe at this point it's easier to keep that import but condition the git checkout script invocation on `if not(parameters.isBuiltFromVmr)`, as you mentioned above?

Reply:
`1.` is better imho as it has fewer implications and the builds would not get different in another detail?

Reply:
I just saw your comment. I explored option 2. You could please take a look at that commit? I think it's cleaner.

Reply:
Oh yeah, it looks actually much simpler now